application_sdk/lock/__init__.py

```diff
@@ -0,0 +1,40 @@
+import json
+import os
+from typing import Callable, Optional
+
+from application_sdk.observability.logger_adaptor import get_logger
+
+logger = get_logger(__name__)
+
+
+def needs_lock(max_locks: int = 5, lock_name: Optional[str] = None):
```

Comment:
work with @junaidrahim to define defaults which should be programmatic vs driven via environment variables.

example: in case the underlying store scales we might need to increase the slots, which should be configurable via environment